### PR TITLE
Added installconfig-upgrade (upgrade details for MAAS)

### PR DESCRIPTION
**Pull request by degville**
- created a brief upgrade doc with troubleshooting (tested multiple upgrades from 14.04)
- added reference from install overview page
- added navigation

### CODE REVIEW

File: src/en/installconfig-upgrade.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+modifications to this file, you can safely allow the upgrade process to
+overwrite the file. But if you have made changes, you'll need to merge your
+changes into the new file manually. 
+
+At the end of the upgrade, if you see processing errors with the `maas`,
+`squid3`, `maas-dns`, or `maas-region-controller` packages, or if the upgrade
+complains about unmet dependencies with the MAAS packages, try the following:
+
+```bash
+sudo apt-get -f install
```
Comment (pmatulis) [inline src/en/installconfig-upgrade.md]: just 'sudo apt -f install' ?

File: src/en/installconfig-upgrade.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+See the '[Upgrading from Ubuntu 14.04 LTS or
+15.10](https://wiki.ubuntu.com/XenialXerus/ReleaseNotes#Upgrading_from_Ubuntu_14.04_LTS_or_15.10)' 
+section in the release notes for Ubuntu 16.04 LTS (Xenial Xerus) for
+instructions on the upgrade process.
+
+## Troubleshooting
+
+Before you start, make sure you've updated your current Ubuntu series to the
+latest packages. You may also need to add both 'apt' and 'dpkg' packages, if
+not installed already, as these are needed by the upgrade process.
```
Comment (pmatulis) [inline src/en/installconfig-upgrade.md]: This paragraph should probably be better placed in the upper section.

Maybe just:
[...] make sure all your packages are updated (`sudo apt update && sudo apt full-upgrade`).

Did you encounter an Ubuntu system that does *not* have the 'apt' and 'dpkg' packages installed? This should never be the case.

File: src/en/installconfig-upgrade.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
+Title: Upgrade from MAAS 1.9
+
+# Upgrade from MAAS 1.9
+
```
Comment (pmatulis) [inline src/en/installconfig-upgrade.md]: We should recommend backups be made prior to upgrading (link to backup page). Possibly even add a recommendation for upgrading a test environment that mirrors the production environment.

Comment (degville) [inline src/en/installconfig-upgrade.md]: Good idea about the test environment. I'll add that. I was uncertain about backups as it seems so obvious, but there's no harm in adding it.

File: src/navigation.tpl [hunk 4]
```diff
@@ -18,6 +18,7 @@
 				<li><a href="installconfig-server-iso.html">Install from Ubuntu Server ISO</a></li>
 				<li><a href="installconfig-package-install.html">Install from packages</a></li>
 				<li><a href="installconfig-lxd-install.html">Install into LXD</a></li>
+				<li><a href="installconfig-upgrade.html">Upgrade from MAAS 1.9</a></li>
```
Comment (pmatulis) [inline src/navigation.tpl]: I think the link and the visible text can be improved by being explicit that this involves going from 1.9 to 2.0.

`<li><a href="installconfig-upgrade-1.9-to-2.0.html">Upgrade from 1.9 to 2.0</a></li>`

File: src/en/installconfig-upgrade.md (hunk 2 above)
Comment (degville) [inline src/en/installconfig-upgrade.md]: 'apt' wasn't in one of my 14.04 test environments.

File: src/en/installconfig-upgrade.md (hunk 1 above)
Comment (degville) [inline src/en/installconfig-upgrade.md]: Definitely needs 'sudo apt upgrade', as this is after the original upgrade has failed so some packages haven't upgraded.

File: src/navigation.tpl (hunk 4 above)
Comment (degville) [inline src/navigation.tpl]: ok, thanks.

**Comment by pmatulis** [comment]
Review complete. Thanks for this important page.

File: src/en/installconfig-upgrade.md (hunk 1 above)
Comment (pmatulis) [inline src/en/installconfig-upgrade.md]: I was just referring to the '-f install' command (i.e. s/apt-get/apt).

**Comment by degville** [comment]
Updated after feedback from @pmatulis.

File: src/en/installconfig-upgrade-1.9-to-2.0.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+(Trusty Tahr) until they reach [end-of-life](https://wiki.ubuntu.com/Releases).
+
+However, it is possible to upgrade from MAAS 1.9.x to MAAS 2.x as part of the
+distribution upgrade from Ubuntu 14.04 (or Ubuntu 15.10) to Ubuntu 16.04. The
+MAAS database and MAAS configuration will be migrated to the latest MAAS
+version as part of the upgrade process.
+
+Before you start, make sure all your packages are updated (`sudo apt update &&
+sudo apt full-upgrade`). Backups should [also be
+made](https://help.ubuntu.com/14.04/serverguide/backups.html) , and we'd
```
Comment (pmatulis) [inline src/en/installconfig-upgrade-1.9-to-2.0.md]: Ah. I was thinking of a specific MAAS backup. I thought we had a backup.md page. I may have been thinking of Juju.

Comment (degville) [inline src/en/installconfig-upgrade-1.9-to-2.0.md]: I'll remove the link and create a todo for MAAS backup.

**Comment by pmatulis** [comment]
Thanks Graham :space_invader:

**Pull request merged**